### PR TITLE
Transitive induction

### PR DESCRIPTION
**Pull request by Drahflow**
Revised version of #17, now against develop branch.

### CODE REVIEW

File: set.mm [hunk 1]
```diff
@@ -43525,6 +43525,34 @@ same distinct variable group (meaning ` A ` cannot depend on ` x ` ) and
   $}
 
   ${
+    $d x A $.  $d x B $.
+    iuneq1d.1 $e |- ( ph -> A = B ) $.
```
Comment (Drahflow) [inline set.mm]: There is a lot to be said for keeping the antecedent in exactly the same format (cf. `iuneq1dv`) for automated proof generation by recursion.

**Comment by Drahflow** [comment]
Anything left I should do to make this mergeable?

**Comment by digama0** [comment]
Sorry, forgot this was open. Some remarks:

I would suggest against using U. U. R for general relations, in favor of an explicit base set. It is often too much to ask of a non-reflexive relation that every element of the base set is either less or greater than some other element; this is effectively the same as eliminating all isolated points from a graph. In my latest project,  I have found the need to use the n-times composition of a relation, but the U. U. R causes problems because I already know the base set, while the relation may very well be empty (yet I still want ( R ^r 0 ) = ( _I |` B ) ). Might I suggest something like ( <. R , B >. ^r N ) or ( ( R ^r B ) ` N ) instead? (The second one will probably be easier to work with, but makes less sense with the ^r symbol.) In this case you can replace your Rel R /\ R e. V hypothesis with R C_ ( B X. B ) /\ B e. V instead.

You could also extend your ^r to negative exponents using the converse; this gives an easy way to write the reflexive/transitive/symmetric closure as U_ n e. ZZ ( R ^r n ).

After some of my own work is published, there will be a definition for the set of reflexive relations and reflexive/transitive relations on a base set, and these will be proven to be Moore systems, so this gives an alternative characterization for t* in terms of mrCls.

**Comment by nmegill** [comment]
Hi Drahflow,

Can you run 'verify markup */f' (added to the most recent metamath.exe, 
version 0.121) and repair the warnings?  Also, if you run 'write source 
set.mm /rewrap' to standardize the wrapping, it will make future diffs 
more meaningful.  As soon as you do these I'll update the web site.

The new convention on dates (that verify markup will check) is described 
at https://groups.google.com/forum/#!msg/metamath/dpnLKGJCgcs/TxesR8rGCQAJ

Thanks,
Norm

**Comment by Drahflow** [comment]
@nmegill: See #23.

@digama0: So after your work is published, will you already have a definition of ^r with explicit base set? If yes, it'd be a waste of time for me to do another one :P . Or do you need one, so I should rather make something real quick[tm]?

**Comment by digama0** [comment]
No, what I defined was a very particular case of ^r with explicit base set where B is (Base`K) and R is the covering relation in a lattice, ( <o ` K ). I did not attempt to define the general notion as you have done. It's just that I find myself doing a lot of similar theorems to what you have, but it isn't as simple as just importing your version and retrofitting my theorems because of the explicit base set thing. It's not essential for my work to have the general definition (I will just have these repeated theorems in the present version) but I figure that if you are working on a generalization than it should be general enough to handle all cases that might appear in set.mm, which is why I mentioned it.

**Comment by nmegill** [comment]
Just to let you know, all new (dummy) variables on the right-hand side
of a definition must have $d's.  I added them into

   df-relexp
   df-rtrclrec

and also to several theorems that they propagated to.  This was done in 
develop commit d9b26f8.

Definition soundness can be verified by mmj2.  You can turn on the test 
by adding

 
SetMMDefinitionsCheckWithExclusions,ax-*,df-bi,df-clab,df-cleq,df-clel,df-sbc

to your RunParms.txt file.

Norm

**Pull request merged**